### ISSUE TITLE
fn state_pruning does not return `PruningMode::Default`.

### ISSUE THREAD

**Issue by maciejnems** [paritytech/substrate#13273]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

`CliConfiguration` trait does not return `PruningMode::default()` for [fn state_pruning](https://github.com/paritytech/substrate/blob/a9c4334d96cbb63aff8e3141a5b3dcad8c636bb7/client/cli/src/config.rs#L245) as stated in docs: 
```
	/// Get the state pruning mode.
	///
	/// By default this is retrieved from `PruningMode` if it is available. Otherwise its
	/// `PruningMode::default()`.
	fn state_pruning(&self) -> Result<Option<PruningMode>> {
		self.pruning_params()
			.map(|x| x.state_pruning())
			.unwrap_or_else(|| Ok(Default::default()))
	}
```
It returns `Ok(Default::default())`, which is `Ok(Option::default())`, so in the end it is `Ok(None)`.

Default `256` value for state pruning is from using `.unwrap_or_default()` in `StateDb::open(...)`: https://github.com/paritytech/substrate/blob/a9c4334d96cbb63aff8e3141a5b3dcad8c636bb7/client/state-db/src/lib.rs#L538

So `state_pruning` function should instead look like:
```
	fn state_pruning(&self) -> Result<Option<PruningMode>> {
		self.pruning_params()
			.map(|x| x.state_pruning())
			.unwrap_or_else(|| Ok(Some(Default::default())))
	}
```

### Steps to reproduce

In `bin/node-template/node/src/command.rs` ([here](https://github.com/paritytech/substrate/blob/a9c4334d96cbb63aff8e3141a5b3dcad8c636bb7/bin/node-template/node/src/command.rs#L205)), right after
```
let runner = cli.create_runner(&cli.run)?;
```
add 
```
println!("{:?}", cli.run.state_pruning());
```
and run node without providing pruning parameter.

**Comment by bkchr** [comment]
Yeah, we should update the docs, but otherwise the implementation is currently correct. We need to return `None` when you don't specify any pruning mode to fetch the pruning mode from the database.
